packages/core/src/docdb/commands/renameCluster.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { getLogger, sleep, ToolkitError } from '../../shared'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { showViewLogsMessage } from '../../shared/utilities/messages'
+import { validateClusterName } from '../utils'
+import { DBClusterNode } from '../explorer/dbClusterNode'
+import { DBGlobalClusterNode } from '../explorer/dbGlobalClusterNode'
+import { telemetry } from '../../shared/telemetry'
+
+/**
+ * Renames a DocumentDB cluster.
+ *
+ * Prompts the user for the new cluster name
+ * Updates the cluster.
+ * Refreshes the node.
+ */
+export async function renameCluster(node: DBClusterNode | DBGlobalClusterNode) {
+    getLogger().debug('docdb:RenameCluster called for: %O', node)
+
+    await telemetry.docdb_renameCluster.run(async () => {
+        if (!node) {
+            throw new ToolkitError('No node specified for RenameCluster')
+        }
+
+        const clusterName = node.name
+
+        if (node.cluster.Status !== 'available') {
+            void vscode.window.showErrorMessage(
+                localize('AWS.docdb.deleteCluster.clusterStopped', 'Cluster must be running')
+            )
+            throw new ToolkitError('Cluster not available', { cancelled: true, code: 'docdbClusterStopped' })
+        }
+
+        const newClusterName = await vscode.window.showInputBox({
+            prompt: localize('AWS.docdb.renameCluster.prompt', 'New cluster name'),
+            value: clusterName,
+            validateInput: validateClusterName,
+        })
+
+        if (!newClusterName) {
+            getLogger().debug('docdb:RenameCluster cancelled')
+            throw new ToolkitError('User cancelled renameCluster', { cancelled: true })
+        }
+
+        try {
+            const cluster = await node.renameCluster(newClusterName)
+
+            getLogger().info('docdb:Renamed cluster: %O', cluster)
+            void vscode.window.showInformationMessage(
+                localize('AWS.docdb.renameCluster.success', 'Updated cluster: {0}', clusterName)
+            )
+
+            await sleep(1000) // wait for server to update status
```

Comment:
Is there any other way we can do this rather than just doing a sleep?

We have a util function called waitUntil that allows us to wait until some event happens. Can we use that to wait until the cluster updates the status before returning?

Reply:
@jpinkney-aws I was originally using waitUntil but the rename operation causes a 404 status later as the resource identifier is changing.  Manual testing revealed the server doesn't update the status for several seconds.